fix: Improve error handling during repository cloning #8293

imadx opened the pull request:
## 🧢 Changes

Use shared parseError to normalize error objects before extracting messages for the repository cloning screen. Construct a clearer display string that includes the error name when it differs from the message, falling back to the parsed message
otherwise.

This replaces ad-hoc checks for message properties and ensures consistent, readable error text for clone failures.

## 🎫 Affected issues

Fixes: https://github.com/gitbutlerapp/gitbutler/issues/8293

<img width="679" height="528" alt="image" src="https://github.com/user-attachments/assets/a0677d11-4682-45fb-8146-0f06ed88fe43" />

imadx committed:
fix: Improve error handling during repository cloning #8293

Byron reviewed (approved):
Thanks so much, this looks so much better as well!

Could @Caleb-T-Owens have a final look before merging?

imadx committed:
fix(ui): render clone form item labels safely

Replace raw HTML insertion with a plain span wrapper when rendering
CloneForm item labels. This removes the use of {@html item.label}
and prevents unescaped HTML from being injected into the DOM,
mitigating XSS risk and ensuring labels are treated as text.

Keep rendering logic otherwise unchanged; preserve item order and
conditional checks.

krlvi reviewed (approved):
Thank you for this contribution @imadx
